src/promptflow/promptflow/_sdk/entities/_run.py

```diff
@@ -486,36 +488,63 @@ def _to_rest_object(self):
                         )
                     inputs_mapping[k] = val
 
-        if self._use_remote_flow:
-            # submit with params flow_definition_resource_id which will be resolved in pfazure run create operation
-            # the flow resource id looks like: "azureml://locations/<region>/workspaces/<ws-name>/flows/<flow-name>"
-            if not isinstance(self.flow, str) or (
-                not self.flow.startswith(FLOW_RESOURCE_ID_PREFIX) and not self.flow.startswith(REGISTRY_URI_PREFIX)
-            ):
-                raise ValueError(
-                    f"Invalid flow value when transforming to rest object: {self.flow!r}. "
-                    f"Expecting a flow definition resource id starts with '{FLOW_RESOURCE_ID_PREFIX}' "
-                    f"or a flow registry uri starts with '{REGISTRY_URI_PREFIX}'"
+        if str(self.flow).startswith(REMOTE_URI_PREFIX):
+            if not self._use_remote_flow:
+                # in normal case, we will upload local flow to datastore and resolve the self.flow to be remote uri
+                # upload via _check_and_upload_path
+                # submit with params FlowDefinitionDataStoreName and FlowDefinitionBlobPath
+                path_uri = AzureMLDatastorePathUri(str(self.flow))
+                return SubmitBulkRunRequest(
```

Comment:
can we merge the return statement? seems that most of the arguments are the same

Reply:
Got confused on what exact change we have done to fix

Reply:
Seems `self._use_remote_flow` only affect `flow_definition_data_store_name, flow_definition_blob_path, flow_definition_resource_id`. 
Can we introduce 3 parameters to avoid the duplicated code?

Reply:
Used partial function to avoid duplicate code.